packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -139,287 +138,377 @@ function readFile(filename, encoding) {
-        memo[x[attrKey]][x[attrKey2]] = x;
-        return memo;
-      }, {});
-
-    //TODO seems like a confusing namespace convention to put id lookups under kind lookups
-    //TODO maybe these should be maps instead of POJO
-    introspectionResultsByKind.namespaceById = xByY(
-      introspectionResultsByKind.namespace,
-      "id"
-    );
-    introspectionResultsByKind.classById = xByY(
-      introspectionResultsByKind.class,
-      "id"
-    );
-    introspectionResultsByKind.typeById = xByY(
-      introspectionResultsByKind.type,
-      "id"
-    );
-    //TODO maybe just add these straight to the class objects
-    introspectionResultsByKind.attributeByClassIdAndNum = xByYAndZ(
-      introspectionResultsByKind.attribute,
-      "classId",
-      "num"
-    );
-    introspectionResultsByKind.extensionById = xByY(
-      introspectionResultsByKind.extension,
-      "id"
-    );
-
-
-    /**
-     *
-     *
-     * @param array
-     * @param newAttr
-     * @param lookupAttr
-     * @param lookup
-     * @param missingOk
-     */
-    const relate = (array, newAttr, lookupAttr, lookup, missingOk = false) => {
-      array.forEach(entry => {
-
-        //relationship
-        const key = entry[lookupAttr];
-
-        if (Array.isArray(key)) {
-          entry[newAttr] = key
-            .map(innerKey => {
-              const result = lookup[innerKey];
-              if (innerKey && !result) {
-                if (missingOk) {
-                  return;
-                }
-                throw new Error(
-                  `Could not look up '${newAttr}' by '${lookupAttr}' ('${innerKey}') on '${JSON.stringify(
-                    entry
-                  )}'`
-                );
-              }
-              return result;
-            })
-            .filter(_ => _);
-        } else {
-          const result = lookup[key];
-          if (key && !result) {
-            if (missingOk) {
-              return;
-            }
-            throw new Error(
-              `Could not look up '${newAttr}' by '${lookupAttr}' on '${JSON.stringify(
-                entry
-              )}'`
-            );
-          }
-          entry[newAttr] = result;
-        }
-      });
-    };
-
-    relate(
-      introspectionResultsByKind.class,
-      "namespace",
-      "namespaceId",
-      introspectionResultsByKind.namespaceById,
-      true // Because it could be a type defined in a different namespace - which is fine so long as we don't allow querying it directly
-    );
-
-    relate(
-      introspectionResultsByKind.class,
-      "type",
-      "typeId",
-      introspectionResultsByKind.typeById
-    );
-
-    relate(
-      introspectionResultsByKind.attribute,
-      "class",
-      "classId",
-      introspectionResultsByKind.classById
-    );
-
-    relate(
-      introspectionResultsByKind.attribute,
-      "type",
-      "typeId",
-      introspectionResultsByKind.typeById
+      },
+      {}
     );
+  });
```

Comment:
I prefer explicit over implicit; if someone searches the codebase for "namespaceById" I want them to find where it's defined. I don't believe the small reduction in lines of code here justifies the reduction in searchability and explicitness.

Reply:
I think you are absolutely right. I am going to change it back.